platform/core/src/services/HangingProtocolService/HangingProtocolService.ts

```diff
@@ -856,26 +874,39 @@ export default class HangingProtocolService extends PubSubService {
     options = null as HangingProtocol.SetProtocolOptions
   ) {
     const stages = this.protocol.stages;
+    let failureCounter = 0;
     for (let i = 0; i < stages.length; i++) {
-      const stage = stages[i];
+      try {
+        const stage = stages[i];
 
-      const { matchedViewports } = this._matchAllViewports(
-        stage,
-        options,
-        new Map()
-      );
-      const activation = stage.stageActivation || {};
-      if (this.matchActivation(matchedViewports, activation.passive, 0)) {
-        if (this.matchActivation(matchedViewports, activation.enabled, 1)) {
-          stage.status = 'enabled';
+        const { matchedViewports } = this._matchAllViewports(
+          stage,
+          options,
+          new Map()
+        ) || {};
+        const activation = stage.stageActivation || {};
+
+        if (this.matchActivation(matchedViewports, activation.passive, 0)) {
+          if (this.matchActivation(matchedViewports, activation.enabled, 1)) {
+            stage.status = 'enabled';
+          } else {
+            stage.status = 'passive';
+          }
         } else {
-          stage.status = 'passive';
+          stage.status = 'disabled';
         }
-      } else {
-        stage.status = 'disabled';
+      } catch (e) {
+        stages[i].status = 'passive';
+        failureCounter++;
+        console.warn(`The hanging protocol viewport is requesting to display the matching displaysets for ${stages[i]} but something went wrong and it could not be matched.`);
       }
     }
 
+    // it means there is no valid stage
+    if (failureCounter === stages.length) {
```

Comment:
Here you can just iterate over the stages and see if there is a passive or enabled one, rather than throwing.